builtInFunctions/esdtNFTCreate.go

```diff
@@ -319,54 +316,63 @@ func (e *esdtNFTCreate) getESDTNFTCreateInput(
 	args := vmInput.Arguments
 
 	var uris = originalURIs
+	var esdtType uint32
 	var nonce uint64
 	var originalCreator []byte
 	var err error
 
 	isCrossChainToken := e.crossChainTokenCheckerHandler.IsCrossChainOperation(tokenID)
 	if !isCrossChainToken {
+		esdtType, err = e.getTokenType(tokenID)
+		if err != nil {
+			return nil, err
+		}
+
 		nonce, err = getLatestNonce(accountWithRoles, tokenID)
 		if err != nil {
 			return nil, err
 		}
 
 		originalCreator = vmInput.CallerAddr
 	} else {
-		nonce, originalCreator, err = getCrossChainTokenNonceAndCreator(args, vmInput.CallType)
+		esdtType, nonce, originalCreator, err = getCrossChainTokenNonceAndCreator(args, vmInput.CallType)
 		if err != nil {
 			return nil, err
 		}
 
-		uris = uris[:len(uris)-2]
+		uris = uris[:len(uris)-3]
 	}
 
 	return &esdtNFTCreateInput{
+		esdtType:              esdtType,
 		nonce:                 nonce,
 		originalCreator:       originalCreator,
 		uris:                  uris,
 		isCrossChainOperation: isCrossChainToken,
 	}, nil
 }
 
-func getCrossChainTokenNonceAndCreator(args [][]byte, callType vm.CallType) (uint64, []byte, error) {
+func getCrossChainTokenNonceAndCreator(args [][]byte, callType vm.CallType) (uint32, uint64, []byte, error) {
```

Comment:
Let's please return a struct here, since we have too many variables that can be mixed/confused on the caller side. 
Also, let's rename this func